### ISSUE TITLE
Rel_Left/Right/Up/Down are not followed when using LAYOUT_LEFT_RIGHT.

### ISSUE THREAD

**Issue by oliv3r** [plantuml-stdlib/C4-PlantUML#174]
With a simple graph like
![LAYOUT_UP_DOWN](https://plantuml.gitlab-static.net/png/U9ora3zlmZ0GnNT_2ZSJaKgyTEe4PWK9KNMEdCGbbXmdii-YKSLtvqv9m2rpDz_pxxtxmpO1b8VOMSPUZAjjR3Hl0ORmBeHNb-9ie8rL3DhNlGFjeAZxJWnMEJAj0pJML6BtjZwIz7dOYqu5q1wbSeSEPPpsnK0D_YLVVyEKpZv6rBjL6CFsbMTJnRTP-iuJI3u-P9Q-yoMgb7rqZV8ZuLKIkkYKo7yOv_o-wGgt8F8kJ1EaPVvaaBy8cVqfSI1qdBGjJ-RSmjnYzg7CFSc8NHaZQg-_O3x3l2T15bLa7Z5Fn1B3DjerTD8R562plW00)
```plantuml

@startuml

!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Container.puml
!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Context.puml

System(sysA, "System A", "System A")
System(sysB, "System B", "System B")

System_Boundary(sysbA, "System Boundary A") {
   Container(conA, "Container A", "Container A")
   Container(conB, "Container B", "Container B")

   Rel_Right(conA, conB, "Rel right")
}

Rel_Left(sysA, sysB, "Rel left")
Rel_Right(sysA, sysB, "Rel right")

@enduml
```

The behavior changes significantly (left becomes up) when using LAYOUT_LEFT_RIGHT.
![LAYOUT_LEFT_RIGHT](https://plantuml.gitlab-static.net/png/U9ora3jlWp0KXNV_2fS9f0OldJe5Htr9LAqe6JfP1jnWoHXaNwj5LVzxRK48JUPkliVVEVU1jXQu0TSfXAwahfLh16u11djBYE6VwK52wofdXQbx3K93MlST6HJNmRInq2XPaTtDvZL8--USTDo2C5vYE-_WKWkJ3g71l-IBBvZIKPwzl-nBbj_TbwnuUdWiumIXjz73NMn7crtZQAfm5gtVoGgYfmyQhTzB5AEzqmqtOy2hLUXH3udu6s6Cb_LZlreW5s6QO5qc5mRwXw3HMUa7yev2A5R8GmjpYzddPMo2xB4VX0ALYm-Opp3l6I3bLS-SOYw8OmpQ2js4E_y2wG2vQ000)
```plantuml

@startuml

!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Container.puml
!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Context.puml

LAYOUT_LEFT_RIGHT()

System(sysA, "System A", "System A")
System(sysB, "System B", "System B")

System_Boundary(sysbA, "System Boundary A") {
   Container(conA, "Container A", "Container A")
   Container(conB, "Container B", "Container B")

   Rel_Right(conA, conB, "Rel right")
}

Rel_Left(sysA, sysB, "Rel left")
Rel_Right(sysA, sysB, "Rel right")

@enduml
```

And so to get the design right, one has to quirkingly use Up/Down to mean Left/Right.
![LEFT_is_UP](https://plantuml.gitlab-static.net/png/U9ora4zFX30GnU_pAIedI5ouUFAqTFsRO3Gh73mr1Uf2KWfffrc9yRlR1XRHFNlhlFczrvc1hK6kqNOIuA9LbRIr80tYOAwJHFDZV6YniQKrGbUzGg4mhlek6IHNthGnMCksJ7PNcnSl5KzPqd63GZk9xPo3jqheUF0V_4k--C0f7RBqxRd8MNPxbxFzu_r37aO0hwE3kz2C9hqamLIHD5Y_enL4VneqMB-NA4PxgsgkHu-Ngz2JxXF99n12blL3jved5s6QO5r6PmRwYw31dz8Dv1nx8LanpFcpoMb4juS67VC5u96R_gZc6ynBUaYATy_C6MVjKmPiXQhzXRy1S56t0m00)
```plantuml

@startuml

!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Container.puml
!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Context.puml

LAYOUT_LEFT_RIGHT()

System(sysA, "System A", "System A")
System(sysB, "System B", "System B")

System_Boundary(sysbA, "System Boundary A") {
   Container(conA, "Container A", "Container A")
   Container(conB, "Container B", "Container B")

   Rel_Up(conA, conB, "Rel right")
}

Rel_Down(sysA, sysB, "Rel left")
Rel_Up(sysA, sysB, "Rel right")

@enduml
```

**Comment by kirchsth** [comment]
Hi @oliv3r,

this is known PlantUML behavior which we cannot directly influence. We could only introduce a new `LAYOUT_LEFT_RIGHT_KEEP_DIRECTIONS()` macro and rotate it internally like below

```plantuml

@startuml
!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Container.puml
!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Context.puml

' possible (incomplete) extension start >>>>>>>>>>>>>>>>>>

!global $right_rotated=%false()

!procedure LAYOUT_LEFT_RIGHT_KEEP_DIRECTIONS() 
!global $right_rotated=%true()
left to right direction
!endprocedure

!unquoted procedure Rel_D($from, $to, $label, $techn="", $descr="", $sprite="", $tags="", $link="")
!if ($right_rotated)
$getRel("-RIGHT->>", $from, $to, $label, $techn, $descr, $sprite, $tags, $link)
!else
$getRel("-DOWN->>", $from, $to, $label, $techn, $descr, $sprite, $tags, $link)
!endif
!endprocedure
!unquoted procedure Rel_Down($from, $to, $label, $techn="", $descr="", $sprite="", $tags="", $link="")
!if ($right_rotated)
$getRel("-RIGHT->>", $from, $to, $label, $techn, $descr, $sprite, $tags, $link)
!else
$getRel("-DOWN->>", $from, $to, $label, $techn, $descr, $sprite, $tags, $link)
!endif
!endprocedure

!unquoted procedure Rel_U($from, $to, $label, $techn="", $descr="", $sprite="", $tags="", $link="")
!if ($right_rotated)
$getRel("-LEFT->>", $from, $to, $label, $techn, $descr, $sprite, $tags, $link)
!else
$getRel("-UP->>", $from, $to, $label, $techn, $descr, $sprite, $tags, $link)
!endif
!endprocedure
!unquoted procedure Rel_Up($from, $to, $label, $techn="", $descr="", $sprite="", $tags="", $link="")
!if ($right_rotated)
$getRel("-LEFT->>", $from, $to, $label, $techn, $descr, $sprite, $tags, $link)
!else
$getRel("-UP->>", $from, $to, $label, $techn, $descr, $sprite, $tags, $link)
!endif
!endprocedure

!unquoted procedure Rel_L($from, $to, $label, $techn="", $descr="", $sprite="", $tags="", $link="")
!if ($right_rotated)
$getRel("-DOWN->>", $from, $to, $label, $techn, $descr, $sprite, $tags, $link)
!else
$getRel("-LEFT->>", $from, $to, $label, $techn, $descr, $sprite, $tags, $link)
!endif
!endprocedure
!unquoted procedure Rel_Left($from, $to, $label, $techn="", $descr="", $sprite="", $tags="", $link="")
!if ($right_rotated)
$getRel("-DOWN->>", $from, $to, $label, $techn, $descr, $sprite, $tags, $link)
!else
$getRel("-LEFT->>", $from, $to, $label, $techn, $descr, $sprite, $tags, $link)
!endif
!endprocedure

!unquoted procedure Rel_R($from, $to, $label, $techn="", $descr="", $sprite="", $tags="", $link="")
!if ($right_rotated)
$getRel("-UP->>", $from, $to, $label, $techn, $descr, $sprite, $tags, $link)
!else
$getRel("-RIGHT->>", $from, $to, $label, $techn, $descr, $sprite, $tags, $link)
!endif
!endprocedure
!unquoted procedure Rel_Right($from, $to, $label, $techn="", $descr="", $sprite="", $tags="", $link="")
!if ($right_rotated)
$getRel("-UP->>", $from, $to, $label, $techn, $descr, $sprite, $tags, $link)
!else
$getRel("-RIGHT->>", $from, $to, $label, $techn, $descr, $sprite, $tags, $link)
!endif
!endprocedure

' possible extension end <<<<<<<<<<<<<<<<<<

' now it can be activated or not 
' LAYOUT_LEFT_RIGHT_KEEP_DIRECTIONS()

System(sysA, "System A")
System(sysB, "System B")
System(sysC, "System C")

Rel(sysA, sysB, "Rel")
Rel(sysB, sysC, "Rel")

Person(P,"P")
Person(PU,"P Up")
Person(PD,"P Down")
Person(PL,"P Left")
Person(PR,"P Right")

Rel_Up(P, PU, "Up")
Rel_Down(P, PD, "Down")
Rel_Left(P, PL, "Left")
Rel_Right(P, PR, "Right")
@enduml
```

Without commented `LAYOUT_LEFT_RIGHT_KEEP_DIRECTIONS()` it looks like 
![](http://www.plantuml.com/plantuml/png/xLPDRvj04BtpAoQMgs8fZYyzLKgKn7RRg3H1nAZg2QrXRA-wtgMxGzFy-yuQuey3LLEvdEo3jVCUynuy1jXhHy9IjLR1cTIvgWg45L7ftez6LZnTB2MjgddbqEP64sgwoCrwL2gXVS_GKQ7aV3H-Dumzb7w9HclX22r3sPWxXDHeByh_fe-_g5OFta9fd9DpXS0WQYUD1XZ-yHS4Pqjbva91tyhbYZ9hI10MbsyMGZaC1tn0QKsEHMKHefjl3-aiYwOVPbboz_7JBFiydSRPv2wPZcTt3_UFuG1Q1SbMNa_XWe0CR5WefCMSfD716UfYwyIkbVvH6Mw4dNs2Afk4_OKrwtFea-4_9UQeV87vIb_sUhmiqEMsNhhIIi9wJMBfwfMI-Zkl1dm_5X0UdkOWw2-Hs2Zi3JTNEBowufvspnV3hLjZrVYm2NAEUwgJXw_tHn3LXLmSPjOUcNdIfzJQKckDBUqkCtwWZXzP6dSxPcbv2knr4nPr5vX_VeuUcByBtSvOX0iwXVRQEKkwYooDZnwO_w1qF6O9NrxBd9r2-yTz9hTfymII81SQvWWY9_dJHm76CaN0H_p51ZC87fyTuJfqp-xc77fr1JSSyuwutH6t1yHuHuoP23YjHgZfOe3n1cR0zspX84RhZ0xZyrxCvKkLSWbfkOTCFEBtNNjOv37_0jl34ezjfishrrzHLWUMXDv6SRk1y-Y4qKPr-pRqUCHuexmRNqyaJBp8NwCkghNw3G00)

With called `LAYOUT_LEFT_RIGHT_KEEP_DIRECTIONS()` it looks like 
![](http://www.plantuml.com/plantuml/png/xLPDRvj04BtpAoQMgs8fZYyzLKgKn7RRg3H1nAZg2QrXRA-wtgMxGzFy-yuQuey3LLEvdEo3jVCUynuy1jXhHy9IjLR1cTIvgWg45L7ftez6LZnTB2MjgddbqEP64sgwoCrwL2gXVS_GKQ7aV3H-Dumzb7w9HclX22r3sPWxXDHeByh_fe-_g5OFta9fd9DpXS0WQYUD1XZ-yHS4Pqjbva91tyhbYZ9hI10MbsyMGZaC1tn0QKsEHMKHefjl3-aiYwOVPbboz_7JBFiydSRPv2wPZcTt3_UFuG1Q1SbMNa_XWe0CR5WefCMS-4I3CzJ5rebTA_sZCjm8E_i4LJO9-mjhrkVG9yD_IinH-GBpbRxizNXPeCjjlNIbbOJrciJIrIibzNTU3VX-B20yFCr1q5yYiL7O6swkS7XrnJtjdY-6MxV6gl5X4-GSzrGd3r_lZo2g2xaupAmzClEaJwcrfTOQMzfTPln07J-oDEvspDBo5DZh9YpgBZ3__1mzC7yNkfsn21Tq2ksrSvPq5raQ7pqm_q7fUCmIlhoMEJk5ze_xJ6xJvWaaGIuqp154JlAdZmACPOeW-8ljPH0yFZl2TUYUtSqvzEeARZZa7N6x8suFYF6E63CHS5QDKDF50ECDp83lsS91ZDOP7SRdlPZBbohb4j9o3vbunE-wzh38O_xrjOSb7jlCcbUllw6i3Ym9lOtYTllcqGcZZUhsNUZnYF56UJUydaYOU96_HbrKQ_KR)

@Potherca: should we add an extension like this, or are our macros complex enough? What do you think?

Best regards
Helmut

**Comment by kirchsth** [comment]
@oliv3r: I think it is solved with PR #177. Can you please test it and add your comments?
Thank you and best regards
Hellmut
